presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -99,8 +99,17 @@ private PlanNode getCostBasedJoin(JoinNode joinNode, Context context)
     {
         List<PlanNodeWithCost> possibleJoinNodes = new ArrayList<>();
 
-        addJoinsWithDifferentDistributions(joinNode, possibleJoinNodes, context);
-        addJoinsWithDifferentDistributions(joinNode.flipChildren(), possibleJoinNodes, context);
+        // don't consider right inequality joins because they execute on a single node.
+        if (joinNode.getCriteria().isEmpty() && joinNode.getType() == LEFT) {
+            addJoinsWithDifferentDistributions(joinNode, possibleJoinNodes, context);
```

Comment:
I would replace it with

```
if (canReplicate(joinNode, context)) {
            possibleJoinNodes.add(getJoinNodeWithCost(context, joinNode.withDistributionType(REPLICATED)));
}
```

So it is explicitly visible what `addJoinsWithDifferentDistributions` does in this special case.